public/app/features/panel/panel_ctrl.ts

```diff
@@ -211,8 +214,8 @@ export class PanelCtrl {
     this.height = calculateInnerPanelHeight(this.panel, containerHeight);
   }
 
-  render(payload?: any) {
-    this.events.emit('render', payload);
+  render(payload?: TableModel | TimeSeries[]) {
+    this.events.emit(panelEventRender, payload);
```

Comment:
this is also a PanelEvents event

Reply:
Mentioned above that since this event's payload depends on `TableModel` and `TimeSeries` it can't be moved to @grafana/ui (where PanelEvents is defined) without moving `TableModel` and `TimeSeries` to @grafana/data

Reply:
the render payload should be up to each panel so no hard requirement on TableModel | TimeSeries.